src/components/ClusterNodes/ClusterSelectionList.tsx

```diff
@@ -35,6 +36,9 @@ export default function ClusterSelectionList({
     const [searchApplied, setSearchApplied] = useState(false)
 
     useEffect(() => {
+        if (window._env_.HIDE_DEFAULT_CLUSTER) {
+            clusterOptions = clusterOptions.filter((item) => item.id !== DEFAULT_CLUSTER_ID)
```

Comment:
`clusterList` is a state variable, we should not update the state variable in this way